ui/src/elements/ngm-object-information.ts

```diff
@@ -1,119 +1,119 @@
-              </button>
-            </div>
-            ${unsafeHTML(it.content)}
-          </div>`) :
-        html`
-          <table class="ui compact small very basic table ngm-info-table">
-            <tbody>
-            ${this.info.properties!.map(row => {
-              const key = row[0];
-              const value = row[1];
-              if ((typeof value === 'string') && (value.startsWith('http'))) {
-                return html`
-                  <tr class="top aligned">
-                    <td class="key">${i18next.t(`assets:${key}`)}</td>
-                    <td class="value"><a href="${value}" target="_blank" rel="noopener">${value.split('/').pop()}</a>
-                    </td>
-                  </tr>
-                `;
-              } else {
-                return html`
-                  <tr class="top aligned">
-                    <td class="key">${key.includes(' ') ? key : i18next.t(`assets:${key}`)}</td>
-                    <td class="value">${value}</td>
-                  </tr>
-                `;
-              }
-            })}
-            </tbody>
-          </table>`;
-
-      if (this.opened && this.info.onshow) {
-        this.info.onshow();
-      }
-      if (!this.opened && this.info.onhide) {
-        this.info.onhide();
-      }
-
-      return html`
-        <div class="ngm-floating-window-header drag-handle">
-          <div class="ngm-close-icon" @click=${() => this.opened = false}></div>
-        </div>
-        <div class="htmlpopup-header" ?hidden="${!this.info.zoom}">
-          <button @click="${this.info.zoom}" class="ui button ngm-zoom-obj-btn ngm-action-btn">
-            ${i18next.t('obj_info_zoom_to_object_btn_label')}
-            <div class="ngm-zoom-plus-icon"></div>
-          </button>
-        </div>
-        <div class="content-container">
-          ${content}
-        </div>
-        ${dragArea}
-      `;
-    } else {
-      return html``;
-    }
-  }
-
-  createRenderRoot() {
-    // no shadow dom
-    return this;
-  }
-}
+import {html} from 'lit';
+import draggable from './draggable';
+import i18next from 'i18next';
+import {LitElementI18n} from '../i18n';
+import {unsafeHTML} from 'lit/directives/unsafe-html.js';
+import QueryStore from '../store/query';
+import {dragArea} from './helperElements';
+import {customElement, property} from 'lit/decorators.js';
+import type {QueryResult} from '../query/types';
+
+
+@customElement('ngm-object-information')
+export class NgmObjectInformation extends LitElementI18n {
+  @property({type: Object})
+  accessor info: QueryResult | undefined;
+  @property({type: Boolean})
+  accessor opened: boolean;
+
+  constructor() {
+    super();
+    this.opened = false;
+    this.hidden = !this.opened;
+
+    QueryStore.objectInfo.subscribe(info => {
+      if (info) {
+        this.info = info;
+        this.opened = true;
+      } else if (this.opened) {
+        this.opened = false;
+        this.info = undefined;

```

Comment:
This could be simplified much further.
```suggestion
      this.info = info;
      this.opened = info == null;
```

Reply:
I had your refactored version before, but since this is an` else if `and not just an `else`, it is not quite equivalent.
It may never occur, but in the new version` this.info` is set to `undefined `if `info `is `undefined `and` this.opened `is `false`. In the original version, `this.info` will only be set to `undefined`, if `this.opened` is `true`.
This case may never occur, but I am not sure of it, which is why i did not simplify all the way

Reply:
As far as I can see, the `else if` is a very premature optimization - it is there so that `info` and `opened` do no get set to the same values that they already contain.